### DISCUSSION TITLE
Problem while generating scaffolding code

### DISCUSSION THREAD

**Discussion by jdarshan5** [callstack/react-native-builder-bob#567] [Q&A] [answered]
I am not able to generate codegen scaffolding code after changing type from `modules` to `all`. I have added a fabric component with turbo modules to try and test the new architecture.

package.json
```json
  "codegenConfig": {
    "name": "RNTestNativeModuleSpec",
    "type": "all",
    "jsSrcsDir": "src"
  }
```

TestCameraView.ts file in spec
```js
import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';
import type { ViewProps, HostComponent } from 'react-native';

interface NativeProps extends ViewProps {
  text: string;
}

export default codegenNativeComponent<NativeProps>(
  'TestCameraView'
) as HostComponent<NativeProps>;
```

generated scaffolding code does not contain anything inside of it like this ShadowNodes.cpp
```cpp
#include <react/renderer/components/RNTestNativeModuleSpec/ShadowNodes.h>

namespace facebook::react {
}
```
not just only this but every file is empty like this

**Comment by atlj** [comment, marked as answer]
The problem is `TestCameraView.ts`'s name. You have to append `NativeComponent` so it becomes `TestCameraViewNativeComponent.ts`. Please check this document: https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-libraries-prerequisites.md#fabric-native-components

**Reply by jdarshan5** [reply]
@atlj thanks for pointing out my mistake now the codegen scaffolding code is getting generated. thanks 🙏

**Reply by atlj** [reply]
Also, just a heads-up, we are adding view + module templates soon with #562.
